ROADMAP.md

```diff
@@ -1,25 +1,389 @@
+* Improved testing
+    * End-to-end testing
+    * Conformance tests are passing for all clients
+    * Fuzz tests
+* Improving verification policies
+    * Rego/CUE integration
+    * [Hammurabi from web PKI](https://www.andrew.cmu.edu/user/bparno/papers/hammurabi.pdf)
+* Reduce size of Cosign binary, fewer dependencies
+* Privacy-preserving telemetry (https://divviup.org/)
+
+### Long Term (12+ mo)
+
+* Plumbing and porcelain model for Cosign, to improve Cosign's user experience
+* Policy: sophisticated verification policies, or deeper integration with a policy evaluation service
+* Further integration with TUF through delegations or trusted verification material distribution, and in-toto for signed provenance
+* Adding additional conformance tests to verify consistent behavior across clients, including negative tests
+* Bundle format includes attestation support that interops with SLSA
+* Reference implementation of how to use signature verification across various points in the software development lifecycle, leveraging policy-controller as one example
+
+## Signature Transparency Log (Rekor)
+
+### Short Term (< 6 mo)
+
+* Add 2 independent log monitors
+    * Monitors will only verify log consistency
+* Witnessing
+    * Monitors should transition into fully fledged witnesses
+    * Witnesses should publish signed checkpoints to distributor
+* V1 for GitHub Actions-based identity monitor
+* Timestamp at Entry Upload
+    * Timestamp should be provided by TSA
+    * Timestamp becomes immutable and verifiable
+* Complete architecture document
+* Publish inclusion events to pub/sub queueo
+
+### Medium Term (6-12 mo)
+
+* API v2 Design & Implementation
+    * Reduce API surface to be structured around key types, push parsing logic to clients
+    * Must support witnessing without compromises
+* MVP split-view attack mitigation
+    * Client requests witnessed checkpoints from a set of witnesses or distributors
+    * Client verifies inclusion proof from log using checkpoint size
+    * Offline verifiable proofs include witnessed checkpoints in Bundle
+* Verifiable Search Design
+    * Detecting credential compromise without requiring constant log scraping
+* Design transparent redaction mechanism
+* Reviewing sharding strategy to better align with Certificate Transparency's sharding
+* Experiment with serverless/log tile format for caching and scalability
+* Design log replication mechanism for scaling instances (possibly can use serverless)
+
+### Long Term (12+ mo)
+
+* Verifiable Search Implementation
+* Package repositories participate in mitigating split-views
+    * Witnessed checkpoints attached to artifact signature metadata
+    * Repositories verify log consistency
+* Implementation for log replication
+* Transparent redaction implementation
+* Privacy-preserving log queries, where the log is not aware of client's artifact of interest
+* Roughtime for distributed trusted timestamps
+* Continued fuzzing of large API surface
+
```

Comment:
```suggestion

## Transparency Log Monitor

### Short Term (< 6 mo)

* Initial launch of Sigstore's [log monitor on GitHub Actions](https://github.com/sigstore/rekor-monitor)
   * Monitors acts as both a witness (co-signing checkpoints) and Verifier (monitoring given identities)
   * Co-signed checkpoints are published to GitHub
   * Verifier monitors for OIDC identities and key fingerprints
   * Alerts are filed as GitHub issues
   * Supports sharded log
* At least 2 independent instances of GitHub Actions-based log monitor running and publicly documented
* Publish blog posts on monitoring requirements as an artifact producer
* Create log cloning tool to support offline log monitoring and analysis
   * Can also be used to transform an existing log into a serverless layout

### Medium Term (6-12 mo)

* V1 launch of Sigstore's [log monitor on GitHub Actions](https://github.com/sigstore/rekor-monitor)
   * Witnessed checkpoints are published to distributors, operated by Sigstore and/or community members
   * Support monitoring Fulcio's CT log
   * Alerts are sent in a variety of ways (e.g. email)
   * Productionized and resilient against intermittent failures
* Publish select monitors to status.sigstore.dev

### Long Term (12+ mo)

* Optionality around running witness in different environments, e.g. hardware or cloud
* Integration with other efforts in transparency ecosystems
   * Publishing checkpoints to external community-operated distributors
* Collaborate with CT log monitors (e.g. [sslmate](https://sslmate.com/certspotter/)) to improve identity monitoring

```

Reply:
@trevrosen PTAL and let me know if you see anything that's missing.

Reply:
Looks great - thanks!